internal/cmd/instance/update_sharing_instance_test.go

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package instance
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/Peripli/service-manager-cli/internal/cmd"
+	"github.com/Peripli/service-manager-cli/pkg/smclient/smclientfakes"
+	"github.com/Peripli/service-manager-cli/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+)
+
+var _ = Describe("Update sharing instance command test", func() {
+	var client *smclientfakes.FakeClient
+	var command *UpdateSharingCmd
+	var buffer *bytes.Buffer
+	var instance *types.ServiceInstance
+	var instances *types.ServiceInstances
+	validAsyncUpdateSharingExecution := func(location string, args ...string) *cobra.Command {
+		instance = &types.ServiceInstance{
+			Name: args[0],
+		}
+		for i, arg := range args {
+			if arg == "--id" {
+				instance.ID = args[i+1]
+			}
+		}
+		operation := &types.Operation{
+			State: "in progress",
+		}
+
+		client.StatusReturns(operation, nil)
+		client.ListInstancesReturns(&types.ServiceInstances{ServiceInstances: []types.ServiceInstance{*instance}}, nil)
+		client.UpdateInstanceReturns(instance, location, nil)
+		piCmd := command.Prepare(cmd.SmPrepare)
+		piCmd.SetArgs(args)
+		Expect(piCmd.Execute()).ToNot(HaveOccurred())
+		return piCmd
+	}
+
+	validSyncUpdateSharingExect := func(args ...string) *cobra.Command {
+		return validAsyncUpdateSharingExecution("", append(args, "--mode", "sync")...)
+	}
+
+	invalidUpdateSharingCommandExecution := func(args ...string) error {
+		shCmd := command.Prepare(cmd.SmPrepare)
+		shCmd.SetArgs(args)
+		return shCmd.Execute()
+	}
+	type testCase struct {
+		share        bool
+		title        string
+		asyncMessage string
+		errorMessage string
+	}
+	tests := []testCase{
+		testCase{true, "share",
+			"successfully scheduled for sharing",
```

Comment:
Successfully scheduled for sharing.

Do you need these?

Reply:
no its a test